hoomd/hpmc/UpdaterQuickCompress.cc

```diff
@@ -183,31 +183,37 @@ BoxDim UpdaterQuickCompress::getNewBox(uint64_t timestep)
     hoomd::UniformDistribution<double> uniform(min_scale, 1.0);
     double scale = uniform(rng);
 
-    const auto& target_box = *m_target_box;
+    const auto& target_box_dims = (*m_target_box)(timestep);
```

Comment:
```suggestion
```
Add `const BoxDim& target_box` to the arguments of `getNewBox`. Also, undo all the changes to `getNewBox` below that are no longer necessary as `target_box` is now set.